src/main/java/nextstep/subway/line/domain/Section.java

```diff
@@ -24,49 +30,68 @@
 
     private int distance;
 
-    public Section() {
+    @Builder
+    public Section(final Long id, final Station upStation, final Station downStation, final int distance) {
+        this.id = id;
+        this.registerStation(upStation, downStation, distance);
     }
 
-    public Section(Line line, Station upStation, Station downStation, int distance) {
-        this.line = line;
+    public void registerStation(final Station upStation, final Station downStation, final int distance) {
         this.upStation = upStation;
         this.downStation = downStation;
         this.distance = distance;
     }
 
-    public Long getId() {
-        return id;
+    public void registerLine(Line line) {
+        Optional.ofNullable(line).ifPresent(it -> {
+            this.line = line;
+            it.addSection(this);
+        });
     }
 
-    public Line getLine() {
-        return line;
+    protected boolean isBefore(Section section) {
+        return Objects.equals(downStation, section.getUpStation());
     }
 
-    public Station getUpStation() {
-        return upStation;
+    protected boolean isAfter(Section section) {
+        return Objects.equals(upStation, section.getDownStation());
     }
 
-    public Station getDownStation() {
-        return downStation;
+    protected boolean contains(Station station) {
+        return Objects.equals(upStation, station) || Objects.equals(downStation, station);
     }
 
-    public int getDistance() {
-        return distance;
+    protected boolean hasSameUpStation(Section section) {
+        return Objects.equals(upStation, section.getUpStation());
     }
 
-    public void updateUpStation(Station station, int newDistance) {
-        if (this.distance <= newDistance) {
-            throw new RuntimeException("역과 역 사이의 거리보다 좁은 거리를 입력해주세요");
-        }
-        this.upStation = station;
-        this.distance -= newDistance;
+    protected boolean hasSameDownStation(Section section) {
+        return Objects.equals(downStation, section.getDownStation());
+    }
+
+    protected void updateUpStation(final Section section) {
+        updateDistance(section);
+
+        this.upStation = section.downStation;
+    }
+
+    protected void updateDownStation(final Section section) {
+        updateDistance(section);
+
+        this.downStation = section.upStation;
     }
 
-    public void updateDownStation(Station station, int newDistance) {
-        if (this.distance <= newDistance) {
-            throw new RuntimeException("역과 역 사이의 거리보다 좁은 거리를 입력해주세요");
+    private void updateDistance(final Section section) {
+        int distance = this.distance - section.getDistance();
+        if (distance < 1) {
```

Comment:
거리를 직접 연산하기 보다는, 별도 메소드를 추출해 아래와 같이 사용하는건 별로일까요?

좋은 메소드 이름이 떠오르지 않아 이름은 임의로 정해보았습니다.

```suggestion
        if (isUpdateDistance(section)) {
```

Reply:
거리를 연산하는 메소드와 검사하는 메소드를 추가하여 반영하였습니다!